source/slang/slang-check-impl.h

```diff
@@ -633,10 +633,10 @@ namespace Slang
         FunctionDifferentiableLevel getFuncDifferentiableLevel(FunctionDeclBase* func);
 
             /// Get the processed inheritance information for `type`, including all its facets
-        InheritanceInfo getInheritanceInfo(Type* type);
+        InheritanceInfo getInheritanceInfo(Type* type, Facet::DirectnessVal inheritanceDepth);
```

Comment:
I am not sure it is necessary to track inheritanceDepth here.

Reply:
> Instead of hacking this with overloadRank, we should implement the proper logic in resolveOverload to prefer the method from a more derived type. Note that this is necessary not only for constructors but also other member methods.

Once I address the `resolveOverload` logic directly, we should be able to forgo any changes to ` getInheritanceInfo` and related logic